tools/deviceGen/CMakeLists.txt

```diff
@@ -0,0 +1,93 @@
+get_filename_component(SRC_VIDEO_PATH "${CMAKE_CURRENT_SOURCE_DIR}/input" ABSOLUTE)
+get_filename_component(NBL_DEVICE_GEN_INCLUDE_DIR "${CMAKE_CURRENT_BINARY_DIR}/include" ABSOLUTE) # TODO: fix a bug in checking for GENERATED property which doesnt eveluate properly generator expressions so $<CONFIG> fails
+
+set(NBL_DEVICE_GEN_INCLUDE_DIR "${NBL_DEVICE_GEN_INCLUDE_DIR}" CACHE INTERNAL "" FORCE)
+set(NBL_DEVICE_GEN_OUTPUT_DIR "${NBL_DEVICE_GEN_INCLUDE_DIR}/nbl/video")
+
+message(WARNING "NBL_DEVICE_GEN_OUTPUT_DIR: ${NBL_DEVICE_GEN_OUTPUT_DIR}")
+
+# inputs
+set(INPUT_DL_JSON "${SRC_VIDEO_PATH}/device_limits.json")
+set(INPUT_DF_JSON "${SRC_VIDEO_PATH}/device_features.json")
+
+set(INPUT_JSONS
+  "${INPUT_DL_JSON}"
+  "${INPUT_DF_JSON}"
+)
+
+# outputs
+
+set(OUTPUT_DL_TEMPLATE "${NBL_DEVICE_GEN_OUTPUT_DIR}/SPhysicalDeviceLimits_")
+set(OUTPUT_DF_TEMPLATE "${NBL_DEVICE_GEN_OUTPUT_DIR}/SPhysicalDeviceFeatures_")
+set(OUTPUT_DT_TEMPLATE "${NBL_DEVICE_GEN_OUTPUT_DIR}/device_capabilities_traits_")
+set(OUTPUT_DL_MEMBERS "${OUTPUT_DL_TEMPLATE}members.h")
+set(OUTPUT_DL_SUBSET "${OUTPUT_DL_TEMPLATE}subset.h")
+set(OUTPUT_DF_MEMBERS "${OUTPUT_DF_TEMPLATE}members.h")
+set(OUTPUT_DF_UNION "${OUTPUT_DF_TEMPLATE}union.h")
+set(OUTPUT_DF_INTERSECT "${OUTPUT_DF_TEMPLATE}intersect.h")
+set(OUTPUT_DT_MEMBERS "${OUTPUT_DT_TEMPLATE}members.h")
+set(OUTPUT_DT_TESTERS "${OUTPUT_DT_TEMPLATE}testers.h")
+set(OUTPUT_DT_DEFAULTS "${OUTPUT_DT_TEMPLATE}defaults.h")
+
+# this needs to have paths to REAL locations not with "_" for the script gen purpose
+set(OUTPUT_HEADERS 
+"${OUTPUT_DL_MEMBERS}"
+"${OUTPUT_DL_SUBSET}"
+"${OUTPUT_DF_MEMBERS}"
+"${OUTPUT_DF_UNION}"
+"${OUTPUT_DF_INTERSECT}"
+"${OUTPUT_DT_MEMBERS}"
+"${OUTPUT_DT_TESTERS}"
+"${OUTPUT_DT_DEFAULTS}"
+)
+
+# python script
+set(NBL_GEN_PY "${CMAKE_CURRENT_SOURCE_DIR}/gen.py")
+
+# custom command
+set(NBL_COMMAND
```

Comment:
you should use some kind of argparse python library to handle arguments so eg. you pass `-inJsonDL "${INPUT_DL_JSON}"`

Reply:
you have added those to [gen.py](https://github.com/Devsh-Graphics-Programming/Nabla/pull/643/files#diff-f616d789839c4626b1374c0c82bf01e4bcad873788ee702cbe30f1ff50bbaab0)

```python
parser.add_argument("limits_json_path", type=str, help="The path to the device_limits.json file")
parser.add_argument("features_json_path", type=str, help="The path to the device_features.json file")
parser.add_argument("limits_output_path", type=str, help="The output path for SPhysicalDeviceLimits_")
parser.add_argument("features_output_path", type=str, help="The output path for SPhysicalDeviceFeatures_")
parser.add_argument("traits_output_path", type=str, help="The output path for the device_capabilities_traits_members.h")```

you need to explicitly use them with your arguments